File: backend-api/framework/Cits.Domain/OperationLogs/OperationLog.cs
using System.ComponentModel.DataAnnotations;
using Cits.Entities;
using FreeSql.DataAnnotations;

namespace Cits.OperationLogs;

[Index("idx_operation_log_operation_time_id", "OperationTime,Id")]
[Index("idx_operation_log_operator_time_id", "OperatorId,OperationTime,Id")]
[Index("idx_operation_log_module_time_id", "Module,OperationTime,Id")]
[Index("idx_operation_log_status_time_id", "Status,OperationTime,Id")]
public class OperationLog : Entity
{
    [MaxLength(100)]
    public string? Module { get; set; }

    [MaxLength(100)]
    public string? OperationType { get; set; }

    [MaxLength(50)]
    public string? OperatorId { get; set; }

    [MaxLength(100)]
    public string? OperatorName { get; set; }

    [MaxLength(2000)]
    public string? DepartmentPath { get; set; }

    [MaxLength(64)]
    public string? OperationIp { get; set; }

    [MaxLength(500)]
    public string? OperationLocation { get; set; }

    public bool Status { get; set; }

    public DateTime OperationTime { get; set; }

    public long ElapsedMilliseconds { get; set; }

    [MaxLength(500)]
    public string? RequestPath { get; set; }

    [MaxLength(20)]
    public string? RequestMethod { get; set; }

    public string? RequestParameters { get; set; }

    public string? ResponseParameters { get; set; }

    public string? ErrorMessage { get; set; }
}

File: frontend-pc/apps/web-ele/src/api/monitor/operation-log.ts
import { requestClient } from '#/api/request';

export interface GetOperationLogsInput {
  startTime?: string;
  endTime?: string;
  module?: string;
  operationType?: string;
  operatorId?: string;
  status?: boolean;
  cursor?: string;
  cursorTime?: string;
  cursorId?: string;
  maxResultCount?: number;
}

export interface OperationLogCursorItemDto {
  id: string;
  module?: string;
  operationType?: string;
  operatorId?: string;
  operatorName?: string;
  departmentPath?: string;
  operationIp?: string;
  operationLocation?: string;
  status: boolean;
  operationTime: string;
  elapsedMilliseconds: number;
  requestPath?: string;
  requestMethod?: string;
}

export interface OperationLogCursorResultDto {
  items: OperationLogCursorItemDto[];
  hasMore: boolean;
  nextCursor?: string;
  nextCursorTime?: string;
  nextCursorId?: string;
}

export interface OperationLogDetailDto extends OperationLogCursorItemDto {
  requestParameters?: string;
  responseParameters?: string;
  errorMessage?: string;
}

/**
 * 获取操作日志列表 (游标分页)
 */
export async function getOperationLogListApi(params: GetOperationLogsInput) {
  return requestClient.get<OperationLogCursorResultDto>('/operation-log', {
    params,
  });
}

/**
 * 获取操作日志详情
 */
export async function getOperationLogDetailApi(id: string) {
  return requestClient.get<OperationLogDetailDto>(`/operation-log/${id}`);
}
